File: src/main/angular/src/app/model/route-detail-meal.ts
/**
 * ルート詳細食事
 */
export interface IRouteDetailMeal {
  /** ルート詳細ID */
  routeDetailId?
  /** 所要時間 */				
  mealMinutes?
  /** 食事種類区分 */
  mealKindDiv?
  /** 食事種類名称 */
  mealKindDivName?
  /** 食事費用 */
  mealCost?
}

File: src/main/kotlin/net/yttravelers/spotshare/entities/RouteDetailMeal.kt
package net.yttravelers.spotshare.entities

import com.fasterxml.jackson.annotation.JsonIgnore
import javax.persistence.Table
import javax.persistence.*

@Entity
@Table(name = "routeDetailMeal")
class RouteDetailMeal {
    @Id
    var routeDetailId: Int? = null

    @JsonIgnore
    @OneToOne
    @JoinColumn(name = "routeDetailId")
    @MapsId
    lateinit var routeDetail: RouteDetail

    var mealMinutes: Int? = null

    var mealCost: Int? = null

    //TODO: 区分の設計
//    var mealKindDiv:
    //必要？
//    var mealKindDivName:
}